simple_network_sim/network_of_populations.py

```diff
@@ -412,20 +415,30 @@ def getInternalInfectiousContactsInNode(
     :rtype: A dictionary of {age: number of new infections}
     """
     infectiousContacts: Dict[Age, float] = {}
-    for age in mixingMatrix:
-        infectiousContacts[age] = 0
+    for ageTo in mixingMatrix:
+        infectiousContacts[ageTo] = 0
 
-        numSuscept = getSusceptibles(age, currentInternalStateDict)
-        if numSuscept > 0:
-            contacts = 0
-            for ageInf in mixingMatrix[age]:
-                contacts += getInfectious(ageInf, currentInternalStateDict, infectiousStates) * mixingMatrix[ageInf][age]
+        susceptibles = getSusceptibles(ageTo, currentInternalStateDict)
+        totalInAge = getTotalInAge(currentInternalStateDict, ageTo)
 
-            totalInAge = getTotalInAge(currentInternalStateDict, age)
-            if totalInAge > 0.0:
-                infectiousContacts[age] = contacts * (numSuscept / totalInAge) * contactsMultiplier
-            else:
-                infectiousContacts[age] = 0.0
+        if susceptibles > 0 and totalInAge > 0.0:
+            for ageFrom in mixingMatrix[ageTo]:
+
+                if RANDOM:
+                    infectious = getInfectious(ageFrom, currentInternalStateDict, infectiousStates)
+
+                    contactsAverage = mixingMatrix[ageFrom][ageTo] * contactsMultiplier
+                    numberOfContacts = stats.poisson.rvs(contactsAverage, size=int(infectious))
+                    numberOfContacts = numberOfContacts[numberOfContacts > 0]
+
+                    infectiousContacts[ageTo] += np.sum(stats.hypergeom.rvs(int(totalInAge), int(susceptibles), numberOfContacts))
```

Comment:
Why are you using hypergeom here?

Reply:
So this is from my understanding of what the mixing matrix means.
1) If it's "number of contacts with distinct different people", and the value is 3, we need to pick 3 individuals WITHOUT replacement in the population, and see if we select susceptible. 
2) If it's "number of contacts with people, but they could be the same", and the value is 3, we need to pick 3 individuals WITH replacement in the population, and see if we select susceptible. 

Hypergeometric is basically like the binomial distribution but without replacement. I assumed 1) here, but was going to ask about this at the moment of the proper review.

Reply:
So you're saying that it counts the number of distinct contacts rather than total number of contacts (which could include meeting one person more than once), right? I think you're right, I think the question they asked was how many people you've contacted.

Reply:
> We will now ask you to remember who you have been in contact with yesterday, between 5am yesterday and 5am today. We are only interested in direct contacts, which are people who you met in personand with whomyou exchanged at least a few words, or with whomyou had physical contact (e.g.a handshake, embracing, kissing, contact sports).Note that if you only spoke to someone over the phone or internet, they should not be included in this section. 

https://www.medrxiv.org/content/medrxiv/suppl/2020/04/03/2020.03.31.20049023.DC1/2020.03.31.20049023-2.pdf

So the people interviewed in the comix study would tell them a list of people they met the day before.

Reply:
Great, thanks a lot for confirming this.

Reply:
Thanks for checking on this - I was just checking it, and agree.  Hypergeometric is the right choice here for sampling without replacement.